### DISCUSSION TITLE
Ver. 121 Two applications showing a WPF ChromeWebBrowser. It fails on second running app when calling CreateBrowser

### DISCUSSION THREAD

**Discussion by DougUnderwood** [cefsharp/CefSharp#4735] [General]
Error: CreateBrowser call failed, review the CEF log file for more details

So had existing application on 117 having no issues.

Upgraded to 121 and now I see an issue where I have one application running for Playback of a list of Web pages.
When I use a second application to open and design which pages to show I get a failure when creating a WPF Browser.

If I close the other then the running one will work Design or Playback.
When I run them on the same machine at the same time the last application fails on 

void ManagedCefBrowserAdapter::CreateBrowser(IWindowInfo^ windowInfo, IBrowserSettings^ browserSettings, IRequestContext^ requestContext, String^ address)

Fails here

           if (!CefBrowserHost::CreateBrowser(*winInfo->GetWindowInfo(), _clientAdapter.get(), addressNative,
                *bSettings->_browserSettings, extraInfo, requestCtx))
            {
                throw gcnew InvalidOperationException("CefBrowserHost::CreateBrowser call failed, review the CEF log file for more details.");
            }

Any insight welcome.

Thanks

**Comment by amaitland** [comment]
> Upgraded to 121 and now I see an issue where I have one application running for Playback of a list of Web pages.
> When I use a second application to open and design which pages to show I get a failure when creating a WPF Browser.

Starting in version 120 `CEF` now implements the `Chromium Process Singleton`. Running two process instances with the same cache path (or no RootCachePath) is no longer possible.

See https://github.com/cefsharp/CefSharp/issues/4668 for further details

> When I run them on the same machine at the same time the last application fails on

Almost certainly `Cef.Initialzie` returned false

https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

**Comment by DougUnderwood** [comment]
Thank you that worked for the two instances issue.

121 does not seem stable since my "Player" application runs through various Web Pages every 15 seconds. 
After around a minute or so the application fails and I find errors in the Event Viewer.
Note: This isn't the case with 119.

Faulting application name: CefSharp.BrowserSubprocess.exe, version: 121.3.70.0, time stamp: 0xd15b5718
Faulting module name: libcef.dll, version: 121.3.7.0, time stamp: 0x65c13a97
Exception code: 0x80000003
Fault offset: 0x0000000004881ed4
Faulting process id: 0x1b24
Faulting application start time: 0x01da6e7b566d226a
Faulting application path: C:\Code\Moxie\Release\7.10 R3\binaries\CefSharp.BrowserSubprocess.exe
Faulting module path: C:\Code\Moxie\Release\7.10 R3\binaries\libcef.dll

Thanks,
Doug Underwood

**Comment by amaitland** [comment]
> Faulting module name: libcef.dll, version: 121.3.7.0, time stamp: 0x65c13a97

CEF has it's own issue tracker https://github.com/chromiumembedded/cef

Get a stack trace, see https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#unmanaged-crashes-when-the-process-dies
